test/conformance/channel_tracing_test.go

```diff
@@ -0,0 +1,173 @@
+			})
+		})
+	}
+}
+
+// setupChannelTracing is the general setup for TestChannelTracing. It creates the following:
+// SendEvents (Pod) -> Channel -> Subscription -> K8s Service -> LogEvents (Pod)
+// It returns a string that is expected to be sent by the SendEvents Pod and should be present in
+// the LogEvents Pod logs.
+func setupChannelTracing(t *testing.T, channel string, client *common.Client, loggerPodName string, incomingTraceId bool) string {
+	// Create the Channel.
+	channelName := "ch"
+	channelTypeMeta := common.GetChannelTypeMeta(channel)
+	client.CreateChannelOrFail(channelName, channelTypeMeta)
+
+	// Create the 'sink', a LogEvents Pod and a K8s Service that points to it.
+	pod := resources.EventDetailsPod(loggerPodName)
+	client.CreatePodOrFail(pod, common.WithService(loggerPodName))
+
+	// Create the Subscription linking the Channel to the LogEvents K8s Service.
+	client.CreateSubscriptionOrFail(
+		"sub",
+		channelName,
+		channelTypeMeta,
+		resources.WithSubscriberForSubscription(loggerPodName),
+	)
+
+	// Wait for all test resources to be ready, so that we can start sending events.
+	if err := client.WaitForAllTestResourcesReady(); err != nil {
+		t.Fatalf("Failed to get all test resources ready: %v", err)
+	}
+
+	// Everything is setup to receive an event. Generate a CloudEvent.
+	senderName := "sender"
+	eventID := fmt.Sprintf("%s", uuid.NewUUID())
+	body := fmt.Sprintf("TestChannelTracing %s", eventID)
+	event := &resources.CloudEvent{
+		ID:       eventID,
+		Source:   senderName,
+		Type:     resources.CloudEventDefaultType,
+		Data:     fmt.Sprintf(`{"msg":%q}`, body),
+		Encoding: resources.CloudEventEncodingBinary,
+	}
+
+	// Send the CloudEvent (either with or without tracing inside the SendEvents Pod).
+	sendEvent := client.SendFakeEventToAddressable
+	if incomingTraceId {
+		sendEvent = client.SendFakeEventWithTracingToAddressable
+	}
+	if err := sendEvent(senderName, channelName, channelTypeMeta, event); err != nil {
+		t.Fatalf("Failed to send fake CloudEvent to the channel %q", channelName)
+	}
+	return body
+}
+
+// assertLogContents verifies that loggerPodName's logs contains mustContain. It is used to show
+// that the expected event was sent to the logger Pod.
+func assertLogContents(t *testing.T, client *common.Client, loggerPodName string, mustContain string) {
+	if err := client.CheckLog(loggerPodName, common.CheckerContains(mustContain)); err != nil {
+		t.Fatalf("String %q not found in logs of logger pod %q: %v", mustContain, loggerPodName, err)
+	}
+}
+
+// getTraceID gets the TraceID from loggerPodName's logs. It will also assert that body is present.
+func getTraceID(t *testing.T, client *common.Client, loggerPodName string) string {
+	logs, err := client.GetLog(loggerPodName)
+	if err != nil {
+		t.Fatalf("Error getting logs: %v", err)
+	}
+	// This is the format that the eventdetails image prints headers.
+	re := regexp.MustCompile("\nGot Header X-B3-Traceid: ([a-zA-Z0-9]{32})\n")
```

Comment:
This is a bit hacky...
We should find a better way to get the header, but can be in a different PR..

Reply:
Totally agree, what do you recommend?

Reply:
We can define a library to add and parse headers to/from the logs, each header will be quoted with something like `<header>...</header>`. We can also use the similar way to log and parse the body of cloud event, if we need more accurate check on it.

This requires bigger changes, so it can be in a different PR.

Reply:
https://github.com/knative/eventing/issues/1945 to track.